catroid/src/main/java/org/catrobat/catroid/scratchconverter/WebSocketClient.java

```diff
@@ -0,0 +1,304 @@
+		sendCommand(new AuthenticateCommand(clientID));
+	}
+
+	public void connectAndAuthenticate(final ConnectAuthCallback connectAuthCallback) {
+		this.connectAuthCallback = connectAuthCallback;
+
+		switch (state) {
+			case NOT_CONNECTED:
+				connect(new ConnectCallback() {
+					@Override
+					public void onSuccess() {
+						Log.i(TAG, "Successfully connected to WebSocket server");
+						authenticate();
+					}
+
+					@Override
+					public void onFailure(ClientException ex) {
+						connectAuthCallback.onConnectionFailure(ex);
+					}
+				});
+				break;
+
+			case CONNECTED:
+				Log.i(TAG, "Already connected to WebSocket server!");
+				authenticate();
+				break;
+
+			case CONNECTED_AUTHENTICATED:
+				Log.i(TAG, "Already authenticated!");
+				connectAuthCallback.onSuccess(clientID);
+				break;
+		}
+	}
+
+	@Override
+	public void retrieveInfo() {
+		Preconditions.checkState(state == State.CONNECTED_AUTHENTICATED);
+		Preconditions.checkState(clientID != INVALID_CLIENT_ID);
+		sendCommand(new RetrieveInfoCommand());
+	}
+
+	@Override
+	public boolean isJobInProgress(long jobID) {
+		return messageListener.isJobInProgress(jobID);
+	}
+
+	@Override
+	public int getNumberOfJobsInProgress() {
+		return messageListener.getNumberOfJobsInProgress();
+	}
+
+	@Override
+	public void convertProgram(final long jobID, final String title, final WebImage image, final boolean verbose,
+			final boolean force) {
+		Preconditions.checkState(state == State.CONNECTED_AUTHENTICATED);
+		Preconditions.checkState(clientID != INVALID_CLIENT_ID);
+		final Job job = new Job(jobID, title, image);
+
+		if (state != State.CONNECTED_AUTHENTICATED || webSocket == null) {
+			convertCallback.onConversionFailure(job, new ClientException("Not connected!"));
+			return;
+		}
+
+		if (!messageListener.scheduleJob(job, force, convertCallback)) {
+			Log.e(TAG, "Cannot schedule job since another job of the same Scratch program is already running (job ID "
+					+ "is: " + jobID + ")");
+			convertCallback.onConversionFailure(job, new ClientException("Cannot start this job since the job "
+					+ "already exists and is in progress! Set force-flag to true to restart the conversion while it "
+					+ "is running!"));
+			return;
+		}
+
+		Log.i(TAG, "Scheduling new job with ID: " + jobID);
+		sendCommand(new ScheduleJobCommand(jobID, force, verbose));
+	}
+
+	@Override
+	public void cancelDownload(final long jobID) {
+		Preconditions.checkState(state == State.CONNECTED_AUTHENTICATED);
+		Preconditions.checkState(clientID != INVALID_CLIENT_ID);
+		sendCommand(new CancelDownloadCommand(jobID));
+	}
+
+	@Override
+	public void onUserCanceledConversion(long jobID) {
+		Preconditions.checkState(state == State.CONNECTED_AUTHENTICATED);
+		Preconditions.checkState(clientID != INVALID_CLIENT_ID);
+		messageListener.onUserCanceledConversion(jobID);
+	}
+
+	@Override
+	public void onBaseMessage(BaseMessage baseMessage) {
+		// case: InfoMessage
```

Comment:
We don't need the comment // case: InfoMessage
This is obvious here (the same applies to some other comments so far)